RFCs/0080-Fn::NumberComparison.md

```diff
@@ -0,0 +1,153 @@
+
+# Motivation
+
+CloudFormation users may want to do different things depending on the condition of a numeric variable. This RFC proposes to support the use case of comparing two numbers to create a boolean that can be used in the Fn::If intrinsic function. 
+
+* Reference:
+    * https://github.com/aws-cloudformation/cfn-language-discussion/issues/80
+
+# Examples
+
+## Example 1
+
+Using `Fn::NumberComparison` in Conditions section to set resource properties
+
+## YAML
+```yaml
+Parameters:
+  stage:
+    Description: Deployment stage
+    Type: String
+    Default: gamma
+    AllowedValues:
+      - gamma
+      - prod
+  retentionInDays:
+    Description: Log retention in days
+    Type: Number
+Conditions:
+  IsProd: !Equals
+    - !Ref 'stage'
+    - prod
+  NotMeetRetentionRequirement: !NumberComparison
+    - !Ref 'retentionInDays'
+    - <=
+    - 365
+  ShouldEnforceLogRetention: !And
+    - !Condition 'IsProd'
+    - !Condition 'NotMeetRetentionRequirement'
+Resources:
+  cwLogGroup:
+    Type: AWS::Logs::LogGroup
+    Properties:
+      LogGroupName: cwLogGroup
+      RetentionInDays: !If
+        - ShouldEnforceLogRetention
+        - 365
+        - !Ref 'retentionInDays'
+```
+
+## JSON 
+```json
+{
+  "Parameters": {
+    "stage": {
+        "Description": "Deployment stage",
+        "Type": "String",
+        "Default": "gamma",
+        "AllowedValues": [
+          "gamma",
+          "prod"
+        ]
+    },
+    "retentionInDays": {
+      "Description": "Log retention in days",
+      "Type": "Number"
+    }
+  },
+  "Conditions": {
+    "IsProd": {
+      "Fn::Equals": [
+        { "Ref": "stage"},
+        "prod"
+      ]
+    },
+    "NotMeetRetentionRequirement": {
+      "Fn::NumberComparison": [
+        {"Ref": "retentionInDays"},
+        "<=",
+        365
+      ]
+    },
+    "ShouldEnforceLogRetention": {
+      "Fn::And": [
+        {"Condition": "IsProd"},
+        {"Condition": "NotMeetRetentionRequirement"}
+      ]
+    }
+  },
+  "Resources": {
+    "cwLogGroup": {
+      "Type" : "AWS::Logs::LogGroup",
+      "Properties" : {
+          "LogGroupName" : "cwLogGroup",
+          "RetentionInDays" : {
+            "Fn::If": [
+              "ShouldEnforceLogRetention",
+              365,
+              {"Ref": "retentionInDays"}
+            ]
+          }
+        }
+    }
+  }
+}
+```
+
+# Limitation
+
+The intrinsic function does not support reference to resource properties. It only supports references to Parameters and Mappings.
+
+# Details
+
+`Fn::NumberComparison` is an intrinsic function that takes a `Number`, a `String` with a comparison operator, and another `Number` and returns a `Boolean` representing the comparison between the first number and the second number. The number could be:
+
+* a hardcoded `Number` in the template
+* `Ref` to a Parameter of type `Number`
+* a `Number` outputted by the use of a Function
```

Comment:
Some of the Functions below (`Fn::Sub `, `Fn::Join`) only output strings, so with the current wording they shouldn't be supported